src/models/solution.rs

```diff
@@ -0,0 +1,31 @@
+use serde::{Deserialize, Serialize};
+use surrealdb::sql::Thing;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Solution {
+    pub id: Option<Thing>,
+
+    pub problem_id: Thing,
+    pub creator: Thing,
+    pub title: String,
+    pub content: String,
+
+    pub created_at: chrono::NaiveDateTime,
+    pub updated_at: chrono::NaiveDateTime,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(crate = "rocket::serde")]
+pub struct SolutionData {
+    pub title: String,
+    pub content: String,
+    pub problem_id: String,
+}
```

Comment:
```suggestion
#[serde(crate = "rocket::serde")]
pub struct SolutionData<'r> {
    pub title: &'r str,
    pub content: &'r str,
    pub problem_id: &'r str,
}
```

Consider use references to improve performance, otherwise, add lifetime definations.